pkg/apis/operator/v1alpha1/config_types.go

```diff
@@ -78,3 +85,67 @@ type ConfigList struct {
 func init() {
 	SchemeBuilder.Register(&Config{}, &ConfigList{})
 }
+
+//IsUpToDateWith return true of config resource reached to its final state
+//The state could be either Success or Error one
+func (c *Config) IsUpToDateWith(version string) bool {
+	cond := c.Status.Conditions
+	if len(cond) == 0 {
+		return false
+	}
+
+	latest := cond[0]
+	return latest.Version == version && (latest.Code == ReadyStatus || latest.Code == ErrorStatus)
+}
+
+//IsMarkedInValid return true of config resource marked as invalid
+func (c *Config) IsMarkedInValid(version string) bool {
```

Comment:
How about turning this into 
1. `config.CurrentStatus` 
2. the caller does `config.CurrentStatus() != ErrorStatus` ?